### PR TITLE
8344461: Additional cleanup in NewThreadAction

### PR DESCRIPTION
**Pull request by stuart-marks**
Convert the RMI NewThreadAction class into a couple static methods on RuntimeUtil. Also remove a couple unnecessary qualified exports from java.base to java.rmi.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8344461](https://bugs.openjdk.org/browse/JDK-8344461): Additional cleanup in NewThreadAction (**Bug** - P4)
 * [JDK-8344867](https://bugs.openjdk.org/browse/JDK-8344867): Cleanup unneeded qualified exports to java.rmi (**Bug** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Contributors
 * Roger Riggs `<rriggs@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22557/head:pull/22557` \
`$ git checkout pull/22557`

Update a local copy of the PR: \
`$ git checkout pull/22557` \
`$ git pull https://git.openjdk.org/jdk.git pull/22557/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22557`

View PR using the GUI difftool: \
`$ git pr show -t 22557`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22557.diff">https://git.openjdk.org/jdk/pull/22557.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22557#issuecomment-2518351703)
</details>

### CODE REVIEW

**Comment by stuart-marks** [comment]
@RogerRiggs Thanks, I'll integrate after internal testing run completes.

File: src/java.rmi/share/classes/sun/rmi/runtime/RuntimeUtil.java [hunk 1]
```diff
@@ -36,10 +36,45 @@
...
+ * inherited from the current thread).
+ *
  * @author      Peter Jones
  **/
 public final class RuntimeUtil {
 
+    /**
+     * Cached reference to the system (root) thread group.
+     */
+    static final ThreadGroup systemThreadGroup;
```
Comment (RogerRiggs) [inline src/java.rmi/share/classes/sun/rmi/runtime/RuntimeUtil.java:60]: Can this be private?  Also userThreadGroup below?

Comment (stuart-marks) [reply src/java.rmi/share/classes/sun/rmi/runtime/RuntimeUtil.java]: Good point. They weren't private in the original, which I just copied over, but I guess we're being a bit more meticulous about making things private these days.

**Review by RogerRiggs — APPROVED** [review]
Thanks for the updates.

**Pull request closed without merging**